src/main/java/org/traccar/protocol/RuptelaProtocolDecoder.java

```diff
@@ -236,7 +236,18 @@ protected Object decode(
                 buf.readUnsignedByte(); // timestamp extension
 
                 if (type == MSG_EXTENDED_RECORDS) {
-                    buf.readUnsignedByte(); // record extension
+                    int b = buf.readUnsignedByte(); // record extension
+                    int noRecordsToMerge = (b & 0xf0) >> 4;
+                    int currentRecord = b & 0x0f;
+
+                    if (currentRecord > 0 && noRecordsToMerge >= currentRecord) {
+                        if (!positions.isEmpty()
+                                && positions.get(positions.size() - 1).getDeviceTime()
+                                   .compareTo(position.getDeviceTime()) == 0) {
+                            position = positions.remove(positions.size() - 1);
```

Comment:
We should probably merge the attributes, right? Otherwise you're actually losing the data.

Reply:
We are not loosing data, as this just checks if it needs to get the latest Position (instead of creating/using the newly created one) and then gets it. Down the line from this, this "old/previous" position, with all the attributes it already has, just gets some new attributes added to it (and then it gets re-added to the positions List). See attached screenshot for example. Previously this IOs were split in two Positions in this instance:
![image](https://github.com/traccar/traccar/assets/858821/b132c07f-fad9-4198-9367-b043e5c239b2)

Reply:
P.S: I am aware that I need to fix the Power decoding so that we have V instead of mV stored, it's coming with some other PR soon-ish :)